### PR TITLE
[Calendar] Default reminder to calendars

### LINKED ISSUE #7216: Default reminder time
As user i want to be able to configure a default reminder time so that whenever i create a new event this reminder is configured.

Duplicate of #1591

**AC**

- There is a setting to configure the default reminder time
- The default value is "no reminder"
- The setting value is used whenever a new event is created.
- The setting is synchronized across all devices.

### Test notes
- [ ] Reminders still working
- [ ] User can create and edit a calendar with multiple default reminders
- [ ] Default reminders field shouldn't appear for creating, editing or accepting shared calendars
- [ ] When creating or editing an event, changing the calendar also set the event reminders
- [ ] The user can add reminders during event creation as normal
- [ ] When selecting a calendar, if it has default reminders, it'll overwrite the previous created reminders
- [ ] When editing an event, if the event had reminders, the editor will not load calendar's default reminders
- [ ] When editing an event, if the event has no reminders, no predefined reminder will be shown
- [ ] When creating an event, the editor will load calendar's default reminders
- [ ] Events invites works as expected

### PR DESCRIPTION
**Pull request by andrehgdias**
This commit adds a new feature that allows users to set multiple default reminders at each calendar, either shared or private. These reminders are unique for each user's calendar.

Closes #7216

### CODE REVIEW

**Review by wrdhub — COMMENTED** [review]
Code is looking good.

My one comment is with shared calendars. If you have a shared calendar with a default reminder time, when someone other than you creates an event on that calendar the default reminder is not added. I think it should be added.

File: src/calendar-app/calendar/gui/CalendarGuiUtils.ts [hunk 1]
```diff
@@ -782,6 +782,15 @@ export function getEventType(
 		return EventType.SHARED_RO
 	}
 
+	/**
+	 * userIsOwner was introduced to keep the current behaviour(v235.240718.0)
```
Comment (wrdhub) [inline src/calendar-app/calendar/gui/CalendarGuiUtils.ts:786]: I would not mention this ( I feel like it doesn't need to be explained /here/ why userIsOwner was introduced, maybe mention it in the commit message ), just document the behavior (which you have done, thank you for that)

**Review by wrdhub — COMMENTED** [review]
I also noticed that if you have a calendar that you have a default reminder time on it and then share that calendar, the default reminder time is still applied but gone from the calendar settings. Then there is no way to change or remove the default reminder time. I would remove any default reminder times when a calendar is shared.

**Review by wrdhub — APPROVED** [review]

**Comment by rih-tutao** [comment]
#7346 was created to deal with the merge conflicts, closing this one.

**Pull request closed without merging**